### PR TITLE
BUG: pct_change wrong result when there are duplicated indices

### LINKED ISSUE #30463: pct_change can't work well with groupby, when fill_method =None
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np
data = pd.DataFrame(np.random.random((10,2)), index=['a', 'b']*5)
data.iloc[1:3,:] = np.nan
```
for pct_change function, when the fill_method = None, it works

```
data.pct_change(1, fill_method=None, limit=1)
	0	1
a	NaN	NaN
b	NaN	NaN
a	NaN	NaN
b	NaN	NaN
a	-0.498169	-0.568501
b	-0.315982	1.340587
a	1.341901	-0.489576
b	0.088594	-0.691063
a	-0.514451	0.054695
b	0.844514	-0.604511
```
but when use it with gourpby , it raise a error

```
data.reset_index().groupby('index')[0].pct_change(1, fill_method=None, limit=1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-18-60898304743e> in <module>
----> 1 data.reset_index().groupby('index')[0].pct_change(1, fill_method=None, limit=1)

~/opt/anaconda3/lib/python3.7/site-packages/pandas/core/groupby/generic.py in pct_change(self, periods, fill_method, limit, freq)
   1344                 )
   1345             )
-> 1346         filled = getattr(self, fill_method)(limit=limit)
   1347         fill_grp = filled.groupby(self.grouper.labels)
   1348         shifted = fill_grp.shift(periods=periods, freq=freq)

TypeError: getattr(): attribute name must be string
```
else, `limit = 0` worked with gourpby, but can't run in pct_change

```
data.reset_index().groupby('index')[0].pct_change(1, fill_method='pad', limit=0)  
0         NaN
1         NaN
2         NaN
3         NaN
4         NaN
5   -0.656739
6    0.601904
7    1.549380
8   -0.471434
9   -0.104398
Name: 0, dtype: float64  
``` 
```
data[0].pct_change(1, fill_method='pad', limit=0)  

~/opt/anaconda3/lib/python3.7/site-packages/pandas/core/missing.py in pad_2d(values, limit, mask, dtype)
    546 
    547     if np.all(values.shape):
--> 548         algos.pad_2d_inplace(values, mask, limit=limit)
    549     else:
    550         # for test coverage

pandas/_libs/algos.pyx in pandas._libs.algos.pad_2d_inplace()

ValueError: Limit must be greater than 0
```

pandas version is 0.25.1

### PR DESCRIPTION
**Pull request by fujiaxiang**
- [x] closes  #30463
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -765,6 +765,7 @@ Numeric
 - Bug in :class:`NumericIndex` construction that caused :class:`UInt64Index` to be casted to :class:`Float64Index` when integers in the ``np.uint64`` range were used to index a :class:`DataFrame` (:issue:`28279`)
 - Bug in :meth:`Series.interpolate` when using method=`index` with an unsorted index, would previously return incorrect results. (:issue:`21037`)
 - Bug in :meth:`DataFrame.round` where a :class:`DataFrame` with a :class:`CategoricalIndex` of :class:`IntervalIndex` columns would incorrectly raise a ``TypeError`` (:issue:`30063`)
+- Bug in :meth:`NDFrame.pct_change` when there are duplicated indices (:issue:`30463`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: I don't think this renders, instead use Series.pct_change and DataFrame.pct_change

**Review by jreback — CHANGES_REQUESTED** [review]
tiny comment, other lgtm. ping on green.

**Comment by jreback** [comment]
@jbrockmendel can you give the tests a quick once over.

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (fujiaxiang) [reply doc/source/whatsnew/v1.0.0.rst]: done

**Review by jreback — APPROVED** [review]
lgtm. @jbrockmendel

**Review by TomAugspurger — APPROVED** [review]
Tests look good. Thanks @fujiaxiang. Can you confirm that #30526 is completely fixed (the groupby stuff too)?

**Comment by jbrockmendel** [comment]
LGTM

**Pull request merged**